capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -236,4 +244,76 @@ public static JsonObject createRequestBodyJson(HttpServletRequest request) throw
     }
     return retrievedObjects;
   }
-}
+
+  public static <T extends BaseEntity> T put(CollectionReference collectionReference, HttpServletRequest request, 
+      HttpServletResponse response, GenericClass<T> genericClass) throws IOException {
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    String id = "";
+    try {
+      id = jsonObject.get("id").getAsString();
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
+      System.err.println("This error was likely caused by a lack of an \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    if (id.length() == 0) {
+      System.err.println("Error caused by either an empty or non-existent \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    Map<String, Object> update = new HashMap<>();
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    boolean containsParameter = false;
+    for (String key : jsonObject.keySet()) {
+      for (Field field: fields) {
+        String type = field.getType().getName();
+        if (field.getName().equals(key)) {
+          containsParameter = true;
+          if (type.equals("int") || type.equals("java.lang.Integer")) {
+            update.put(key, jsonObject.get(key).getAsInt());
+          }
+          if (type.equals("java.lang.String") && !field.getName().equals("id")) {
+            update.put(key, jsonObject.get(key).getAsString());
+          }
+          if (type.equals("boolean") || type.equals("java.lang.Boolean")) {
+            update.put(key, jsonObject.get(key).getAsBoolean());
+          }
```

Comment:
nit: it appears these could be elseifs since type can't be an int and a boolean for instance

Reply:
Good call, I realized this in writing my post method. I used the else-ifs there, and also added a final else to catch any types that we do not support. I'll add that same functionality here.